cache/cache.go

```diff
@@ -64,6 +64,22 @@ func New(client *clientv3.Client, prefix string, opts ...Option) (*Cache, error)
 
 	internalCtx, cancel := context.WithCancel(context.Background())
 
+	// Auto-calculate internal watcher buffer size if not explicitly set.
+	// The internal store watcher is critical infrastructure and should have
+	// sufficient buffer to handle burst traffic without lagging.
```

Comment:
It should but that's for performance, cache should work correctly no matter how big buffer it has. Correctly here doesn't mean effectively, just means it should not fail tests. 

Test should validate correctness not performance, so that's why my surprise in the need to change cache node and not just test code.